godot-core/src/builtin/meta/mod.rs

```diff
@@ -270,3 +270,64 @@ impl PropertyInfo {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct MethodInfo {
+    pub id: i32,
+    pub method_name: StringName,
+    pub class_name: ClassName,
+    pub return_type: PropertyInfo,
+    pub arguments: Vec<PropertyInfo>,
+    pub default_arguments: Vec<Variant>,
+    pub flags: global::MethodFlags,
+}
+
+impl MethodInfo {
+    /// Converts to the FFI type. Keep this object allocated while using that!
+    ///
+    /// # Safety
+    /// This function leaks memory that has to be cleaned up by the caller
+    /// once it is no longer used. Specifically the `arguments` and
+    /// `default_arguments` vectors have to be reconstruced from the pointer
+    /// and length and then dropped / freed.
+    ///
+    /// Each vector can be reconstructed with `Vec::from_raw_parts` since the
+    /// pointers where created with `Vec::into_boxed_slice` which gurantees that
+    /// vector capacity and length are equal.
+    pub fn method_sys(&self) -> sys::GDExtensionMethodInfo {
```

Comment:
It is a bit misleading that this is marked as a `Safety` block, and yet the function is not an unsafe function. Since usually `# Safety` is meant to be the safety invariants that need to be upheld by the function when you call it. Instead the `# Safety` line could probably be removed, or just renamed to something other than `Safety`.

Reply:
@TitanNano i think you marked this resolved just as i made this comment